### ISSUE TITLE
interaction.response.defer() not working correctly

### ISSUE THREAD

**Issue by CDE90** [Rapptz/discord.py#7863]
### Summary

interaction.response.defer() not working correctly (slash commands)

### Reproduction Steps

- Defer an interaction using `interaction.response.defer()`
- Try to respond to the interaction using `interaction.response.send_message(...)`

### Minimal Reproducible Code

```python
@tree.command(guild=discord.Object(id=GUILD_ID))
async def test(interaction: discord.Interaction):
    await interaction.response.defer()

    await asyncio.sleep(3)

    await interaction.response.send_message("Done!")
```

### Expected Results

Interaction to be deferred then the response to change to "Done!"

### Actual Results

```
Ignoring exception in command 'test':
Traceback (most recent call last):
  File "...\lib\site-packages\discord\app_commands\commands.py", line 555, in _invoke_with_namespace
    return await self._callback(interaction, **transformed_values)  # type: ignore
  File "...\test.py", line 18, in test
    await interaction.response.send_message("Done!")
  File "...\lib\site-packages\discord\interactions.py", line 641, in send_message
    raise InteractionResponded(self._parent)
discord.errors.InteractionResponded: This interaction has already been responded to before

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "...\lib\site-packages\discord\app_commands\tree.py", line 1043, in call
    await command._invoke_with_namespace(interaction, namespace)
  File "...\lib\site-packages\discord\app_commands\commands.py", line 573, in _invoke_with_namespace
    raise CommandInvokeError(self, e) from e
discord.app_commands.errors.CommandInvokeError: Command 'test' raised an exception: InteractionResponded: This interaction has already been responded to before
```

### Intents

discord.Intents.default()

### System Information

- Python v3.10.1-final
- discord.py v2.0.0-alpha
    - discord.py pkg_resources: v2.0.0a4070+g723584ca
- aiohttp v3.7.4.post0
- system info: Windows 10 10.0.22000

### Checklist

- [X] I have searched the open issues for duplicates.
- [X] I have shown the entire traceback, if possible.
- [X] I have removed my token from display, if visible.

**Comment by LeoCx1000** [comment]
You can send a response only once.
After you defer you need to send trough the follow-up webhook. `interaction.followup.send`

**Comment by cj-clifton** [comment]
When I set `ephemeral=True` in `interaction.followup.send` others can see the message, is this intended?

**Comment by SebbyLaw** [comment]
> When I set `ephemeral=True` in `interaction.followup.send` others can see the message, is this intended?

The `ephemeral=` kwarg in the `defer()` determines whether or not the followup is ephemeral. This is a discord limitation.
